src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2023,10 +2018,13 @@ public List<String> reloadSeeds()
             return getSeeds();
         }
 
+        tmp.remove(getBroadcastAddressAndPort());
```

Comment:
why this change?  we didn't support self seed in this API and we still don't, but if the only seed is the self seed we now bypass this check?  What is the motivation?

Reply:
@dcapwell 

There is this above that:

````
        if (tmp.isEmpty())
        {
            logger.trace("New seed node list is empty. Not updating seed list.");
            return getSeeds();
        }
````

If we removed local address before - so `tmp` is empty - it would return. But in that case `seeds` would not be changed as it would return prematurely - that is the reason we see this behavior described in the ticket.

But we have to remove local address from `tmp` before going to mutate `seeds` 

````
        // Add the new entries
        seeds.addAll(tmp);
        // Remove the old entries
        seeds.retainAll(tmp);
````

This would add nothing with addAll as tmp is empty and it would remove everything when retaininAll(empty list). So `seeds` would end up empty in the end.